rust/src/cliwrap/kernel_install.rs

```diff
@@ -0,0 +1,73 @@
+// If not running on container continue the current path.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+use anyhow::{anyhow, Result};
+use camino::Utf8Path;
+use cap_std::fs::FileType;
+use cap_std::fs_utf8::Dir as Utf8Dir;
+use cap_std_ext::cap_std;
+use std::process::Command;
+
+/// Primary entrypoint to running our wrapped `kernel-install` handling.
+pub(crate) fn main(argv: &[&str]) -> Result<()> {
+    let mut arg = false;
+    if let Some(k) = argv.get(0) {
+        arg = k.eq(&"add");
+    }
+
+    let modules_path = Utf8Dir::open_ambient_dir("lib/modules", cap_std::ambient_authority())?;
+    //kernel-install is called by kernel-core and kernel-modules cleanup let's make sure we just call dracut once.
+    let mut new_kernel: Option<_> = None;
+
+    //finds which the kernel to remove and which to generate the initramfs for.
+    for entry in modules_path.entries()? {
+        let entry = entry?;
+        let kernel_dir = entry.file_name()?;
+        let kernel_binary_path = Utf8Path::new(&kernel_dir).join("vmlinuz");
+
+        if entry.file_type()? == FileType::dir() {
+            if modules_path.exists(kernel_binary_path) {
+                if arg {
+                    new_kernel = Some(kernel_dir);
+                }
+            } else {
+                new_kernel = None;
+                modules_path.remove_dir_all(kernel_dir)?;
+            }
+        }
+    }
+    if let Some(k) = new_kernel {
+        run_dracut(&k)?;
+    }
+    Ok(())
+}
+
+fn run_dracut(kernel_dir: &str) -> Result<()> {
+    let root_fs = Utf8Dir::open_ambient_dir("/", cap_std::ambient_authority())?;
+    root_fs.create_dir("tmp/dracut")?;
+    let res = Command::new("/usr/libexec/rpm-ostree/wrapped/dracut")
```

Comment:
Hmm...tricky, so this is relying on the cliwrap being enabled.  We probably need to support the non-cliwrap case too; maybe change this to detect if the cliwrap'd version exists?

Or actually...maybe change the cliwrap'd dracut entry point to have an environment variable that allows skipping the wrapping?

The cliwrap stuff is a bit tricky...